src/arkode/arkode_sprkstep_io.c

```diff
@@ -0,0 +1,491 @@
+
+  /* Set default ARKODE infrastructure parameters */
+  retval = arkSetDefaults(ark_mem);
+  if (retval != ARK_SUCCESS)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, "ARKODE::SPRKStep", "SPRKStepSetDefaults",
+                    "Error setting ARKODE infrastructure defaults");
+    return (retval);
+  }
+
+  /* use the default method order */
+  SPRKStepSetOrder(arkode_mem, 0);
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetUseCompensatedSums:
+
+  Turns on/off compensated summation in SPRKStep and ARKODE.
+  ---------------------------------------------------------------*/
+int SPRKStepSetUseCompensatedSums(void* arkode_mem, sunbooleantype onoff)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetUseCompensatedSums",
+                                  &ark_mem, &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (onoff)
+  {
+    arkSetUseCompensatedSums(arkode_mem, SUNTRUE);
+    ark_mem->step = sprkStep_TakeStep_Compensated;
+  }
+  else
+  {
+    arkSetUseCompensatedSums(arkode_mem, SUNFALSE);
+    ark_mem->step = sprkStep_TakeStep;
+  }
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetMethod:
+
+  Specifies the SPRK method
+
+  ** Note in documentation that this should not be called along
+  with SPRKStepSetOrder. **
+  ---------------------------------------------------------------*/
+int SPRKStepSetMethod(void* arkode_mem, ARKodeSPRKStorage sprk_storage)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetMethod", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (step_mem->method)
+  {
+    ARKodeSPRKStorage_Free(step_mem->method);
+    step_mem->method = NULL;
+  }
+
+  step_mem->method = sprk_storage;
+
+  return (ARK_SUCCESS);
+}
+
+/*---------------------------------------------------------------
+  SPRKStepSetMethodName:
+
+  Specifies the SPRK method.
+  ---------------------------------------------------------------*/
+int SPRKStepSetMethodName(void* arkode_mem, const char* method)
+{
+  ARKodeMem ark_mem          = NULL;
+  ARKodeSPRKStepMem step_mem = NULL;
+  int retval                 = 0;
+
+  /* access ARKodeSPRKStepMem structure */
+  retval = sprkStep_AccessStepMem(arkode_mem, "SPRKStepSetMethodName", &ark_mem,
+                                  &step_mem);
+  if (retval != ARK_SUCCESS) { return (retval); }
+
+  if (step_mem->method)
+  {
+    ARKodeSPRKStorage_Free(step_mem->method);
+    step_mem->method = NULL;
+  }
+
+  step_mem->method = ARKodeSPRKStorage_LoadByName(method);
```

Comment:
Is this routine guaranteed to succeed?  If not, then shouldn't we check `if (step_mem->method == NULL)` before returning `ARK_SUCCESS`?